schema.py

```diff
@@ -267,22 +267,17 @@ def is_valid(self, data):
         else:
             return True
 
-    def _raise_error(self, error_type):
-        """Returns string to add to error message based on what the error type
-        is and whether a custom schema name has been defined or not.
+    def _raise_error(self, exception, message, *error_list):
```

Comment:
Is there a case where there is more than one error passed to `error_list`? Right now you only process `error_list[0]`. No need to lock yourself into a design decision (i.e. using up `*args`) if you don't have to.

Also, what if you simply accessed `self._error` in this method instead of passing it in? That would eliminate the need for [`format_error`](https://github.com/keleshev/schema/pull/182/files#diff-f9bdc6e9ac5d7d6096bf607cc6260f9eR285) in `validate` since you could simply house that logic inside this method.

Reply:
This flexibility is necessary since there's at least one instances where the `*error_list` argument isn't given a value so it's necessary to make this parameter optional and when the `*error_list` parameter is defaulted to `None`, validations will pass when they should fail.

On line 389:
`self._raise_error(SchemaError, message)`
No 3rd argument given.

I looked into including the `format_error()` function into `_raise_error()`. Unfortunately there are some instances where `self._error` is accessed directly without being formatted (via `self._error.format(data)`) or set to `None` and other instances where the `*error_list` argument is given a value where `self._error` is neither directly accessed nor formatted.

One line 380 to 381:
`self._raise_error(SchemaError, message,
                                  format_error(self._error))`
A `SchemaError` is raised that formats or sets `self._error` to `None`

On line 395:
`self._raise_error(SchemaError, message, self._error)`
Another `SchemaError` is raised, except `self._error` is accessed directly.

There are also other instances where various schema errors access `self._error` both directly and after formatting it.

On line 376-377:
`self._raise_error(SchemaError, [None] + x.autos,
                                  [self._error] + x.errors)`
`self._error` is neither formatted, set to none, nor just simply accessed directly.

So for the above reasons I didn't want `_raise_error()` to perform `format_error()` on the argument given for `*error_list` every time. And since `format_error()` is both applied and not applied for the same type of schema errors, I can't easily set `format_error()` to only be executed based on the exception that's raised.

The only solution I can think of at the moment is to have `self._raise_error()` take in a fourth argument that "flags" whether `format_error()` should be executed on the `*error_list` argument.

It's at this point that I figured, for now at least, it's simpler to leave `format_error()` external to `self._reaise_error()`